File: go/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
	if haystack == "" && needle == "" {
		return 0
	}
	lh := len(haystack)
	ln := len(needle)
	for i := 0; i < lh; i++ {
		ct := 0
		for j := 0; j < ln; j++ {
			if i+j >= lh {
				return -1
			}
			if needle[j] != haystack[i+j] {
				break
			}
			ct++
		}
		if ct == ln {
			return i
		}
	}
	return -1
}

File: csharp/0028-find-the-index-of-the-first-occurrence-in-a-string.cs
public class Solution
{
    public int StrStr(string haystack, string needle)
    {
        if (needle.Length > haystack.Length) return -1;

        int l = 0;
        int r = needle.Length - 1;

        while (r < haystack.Length)
        {
            bool areEqual = true;
            for (int i = l, j = 0; i <= r; i++, j++)
            {
                if (haystack[i] != needle[j])
                {
                    areEqual = false;
                    break;
                }
            }

            if (areEqual)
            {
                return l;
            }

            l++;
            r++;
        }

        return -1;
    }
}